src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -38,7 +38,7 @@ public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
             var warnings = new List<PackagingLogMessage>();
             foreach (var folder in _folders)
             {
-                var files = builder.GetFiles().Where(t => t.StartsWith(folder));
+                var files = builder.GetFiles().Where(t => t.StartsWith(folder + Path.DirectorySeparatorChar));
```

Comment:
The path separators should be normalized like: https://github.com/NuGet/NuGet.Client/blob/48f67796ff38e98e3dfe4ab55fce000577b74fea/src/NuGet.Core/NuGet.Packaging/Rules/ContentFolderInPackageReferenceProjectRule.cs#L24-L26

Reply:
done